wgpu-core/src/command/transfer.rs

```diff
@@ -19,13 +19,45 @@ use hal::command::CommandBuffer as _;
 use thiserror::Error;
 use wgt::{BufferAddress, BufferUsage, Extent3d, TextureUsage};
 
-use std::iter;
+use std::{fmt, iter};
 
 pub(crate) const BITS_PER_BYTE: u32 = 8;
 
 pub type BufferCopyView = wgt::BufferCopyView<BufferId>;
 pub type TextureCopyView = wgt::TextureCopyView<TextureId>;
 
+#[derive(Clone, Debug)]
+pub enum ResourceSide {
+    Source,
+    Destination,
+}
+
+impl fmt::Display for ResourceSide {
```

Comment:
why not just derive it?

Reply:
Because then it would be the wrong capitalization for the error message

Reply:
sure, but it's an error message, so it's fine. I.e. you can put it into quotes to separate from the rest of the message
but it would save code, and that's a good thing to have

Reply:
Fixed.